sympy/physics/control/tests/test_control_plots.py

```diff
@@ -0,0 +1,258 @@
+    if not matplotlib:
+        skip("Matplotlib not the default backend")
+
+    assert _to_tuple(*pole_zero_numerical_data(tf1)) == \
+        ((), ((-0.24999999999999994+1.3919410907075054j), (-0.24999999999999994-1.3919410907075054j)))
+    assert _to_tuple(*pole_zero_numerical_data(tf2)) == \
+        ((0.0,), ((-0.25+0.3227486121839514j), (-0.25-0.3227486121839514j)))
+    assert _to_tuple(*pole_zero_numerical_data(tf3)) == \
+        ((0.0,), ((-0.5000000000000004+0.8660254037844395j),
+        (-0.5000000000000004-0.8660254037844395j), (0.9999999999999998+0j)))
+    assert _to_tuple(*pole_zero_numerical_data(tf7)) == \
+        (((-0.6722222222222222+0.8776898690157247j), (-0.6722222222222222-0.8776898690157247j)),
+        ((2.220446049250313e-16+1.2797182176061541j), (2.220446049250313e-16-1.2797182176061541j),
+        (-0.7657146670186428+0.5744385024099056j), (-0.7657146670186428-0.5744385024099056j),
+        (0.7657146670186427+0.5744385024099052j), (0.7657146670186427-0.5744385024099052j)))
+    assert _to_tuple(*pole_zero_numerical_data(ser1)) == \
+        ((), (5.0, 0.0, 0.0, 0.0))
+    assert _to_tuple(*pole_zero_numerical_data(par1)) == \
+        ((-5.645751311064592, -0.5000000000000008, -0.3542486889354093),
+        ((-0.24999999999999986+1.3919410907075052j),
+        (-0.24999999999999986-1.3919410907075052j), (-0.2499999999999998+0.32274861218395134j),
+        (-0.2499999999999998-0.32274861218395134j)))
+
+
+def test_bode():
+    if not matplotlib:
+        skip("Matplotlib not the default backend")
+
+    def bode_phase_evalf(system, point):
+        expr = system.to_expr()
+        _w = Dummy("w", real=True)
+        w_expr = expr.subs({system.var: I*_w})
+        return arg(w_expr).subs({_w: point}).evalf()
+
+    def bode_mag_evalf(system, point):
+        expr = system.to_expr()
+        _w = Dummy("w", real=True)
+        w_expr = expr.subs({system.var: I*_w})
+        return 20*log(Abs(w_expr), 10).subs({_w: point}).evalf()
+
+    def test_bode_data(sys):
+        return y_coordinate_equality(bode_magnitude_numerical_data, bode_mag_evalf, sys) \
+            and y_coordinate_equality(bode_phase_numerical_data, bode_phase_evalf, sys)
+
+    assert test_bode_data(tf1)
+    assert test_bode_data(tf2)
+    assert test_bode_data(tf3)
+    assert test_bode_data(tf4)
+    assert test_bode_data(tf5)
+
+
+def test_impulse_response():
+    if not matplotlib:
+        skip("Matplotlib not the default backend")
+
+    assert _to_tuple(*impulse_response_numerical_data(tf1, adaptive=False, nb_of_points=10)) == \
+        ((0.0, 1.1111111111111112, 2.2222222222222223, 3.3333333333333335, 4.444444444444445,
+        5.555555555555555, 6.666666666666667, 7.777777777777779, 8.88888888888889, 10.0),
+        (0.0, 0.544019738507865, 0.01993849743234938, -0.31140243360893216, -0.022852779906491996, 0.1778306498155759,
+        0.01962941084328499, -0.1013115194573652, -0.014975541213105696, 0.0575789724730714))
```

Comment:
This test seems fragile by testing for exact equality with floats.

Reply:
Is there any better alternative?

Reply:
It's better to test that the numbers are close to some precision e.g. `abs(result - expected) < 1e-15` or something like that.

Reply:
Done